src/screens/YoutubePlayerScreen.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
+import { View, SafeAreaView, StyleSheet } from 'react-native';
+import { WebView, WebViewNavigation } from 'react-native-webview';
+import { HomeStackParamList } from '../navigators';
+
+export const videoUrl = (youtubeVideoId: string) => {
+  return `https://www.youtube.com/embed/${youtubeVideoId}?modestbranding=1&playsinline=1&showinfo=0&rel=0&autoplay=1" frameborder="0" style="overflow:hidden;overflow-x:hidden;overflow-y:hidden;height:100%;width:100%;position:absolute;top:0px;left:0px;right:0px;bottom:0px" height="100%" width="100%"`;
+};
+
+export interface Params {
+  youtubeVideoId: string;
+  videoName: string;
+}
+
+export const onShouldStartLoadWithRequest = (
+  videoPlayer: React.RefObject<WebView<{}>>,
+  event: WebViewNavigation,
+) => {
+  if (event.url.indexOf('embed') !== -1 || event.url === 'about:blank') {
+    return true;
+  } else {
+    videoPlayer.current && videoPlayer.current.stopLoading();
```

Comment:
What does this do for us?

Reply:
We used this logic in `life-extend`. I think it's because of this:

> If you want to avoid the the players from being redirected to different URLs from the iframe you can used two methods — onShouldStartLoadWithRequest for iOS and onNavigationStateChange for Android.

Source: https://surajmdurgad.medium.com/embedding-youtube-videos-in-a-react-native-app-8c556a18fd76